test/test_other.py

```diff
@@ -14020,6 +14020,19 @@ def test_prejs_unicode(self):
     ''')
     self.do_runf('hello_world.c', '↓', emcc_args=['--pre-js=script.js'])
 
+  # This test verifies that concatenating input --pre-js style files will not trip up even
+  # if one of the submitted files does not have a newline at the end of the file.
+  @parameterized({
+    'prejs': ('--pre-js',),
+    'postjs': ('--post-js',),
+    'extern_prejs': ('--extern-pre-js',),
+    'extern_postjs': ('--extern-post-js',),
+  })
```

Comment:
Do we really need all 4 variants?

Can you just test pre-js and call this `test_prejs_trailing_newline`?   (to match the name of the test above which also starts with `test_prejs`, and also only tests one of the 4 variants?)

Reply:
> Do we really need all 4 variants?

Well, we don't need any of them if it works. But isn't the point of tests to verify that things won't break in the future? Just wanted to be comprehensive.